File: mvpjava/src/main/java/com/example/mvpjava/MainActivity.java
package com.example.mvpjava;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.example.mvpjava.model.login.LoginActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_to_login).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, LoginActivity.class));
            }
        });
    }
}

File: mvp2/src/main/java/com/example/mvp2/MainActivity.kt
package com.example.mvp2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.mvp2.ui.course.activity.CourseActivity
import com.example.mvp2.ui.login.LoginActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), MainContract.View {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mPresenter = MainPresenter(this)

        btn_to_course.setOnClickListener {
            startActivity(Intent(this, CourseActivity::class.java))
        }

        btn_to_login.setOnClickListener {
            Intent(this, LoginActivity::class.java).run {
                startActivity(this)
            }
        }

        btn_get_data.setOnClickListener {
            mPresenter.loadData()
        }
    }

    /**
     * 实现MainContract.View接口，并实现对应的方法
     */
    override fun onDataLoaded(s: String) {
        tv_data.text = s
    }
}
